src/vs/base/common/fuzzyScorer.ts

```diff
@@ -905,7 +905,8 @@ function normalizeQuery(original: string): { pathNormalized: string; normalized:
 	// - wildcards: are used for fuzzy matching
 	// - whitespace: are used to separate queries
 	// - ellipsis: sometimes used to indicate any path segments
-	const normalized = pathNormalized.replace(/[\*\u2026\s"]/g, '');
+	// - hash: used by some language servers (e.g. rust-analyzer) as query modifiers
+	const normalized = pathNormalized.replace(/[\*\u2026\s"#]/g, '');
```

Comment:
This change influences the Search Service:
* https://github.com/microsoft/vscode/blob/d3ac31d681afb4e3bd12acde65b573fd7f61c557/src/vs/workbench/services/search/node/fileSearch.ts#L78
* https://github.com/microsoft/vscode/blob/d3ac31d681afb4e3bd12acde65b573fd7f61c557/src/vs/workbench/services/search/node/rawSearchService.ts#L261
* https://github.com/microsoft/vscode/blob/d3ac31d681afb4e3bd12acde65b573fd7f61c557/src/vs/workbench/services/search/node/rawSearchService.ts#L322

@roblourens / @osortega can I have you review this?